File: code-playbook/odds_ends/spring-cloud-msa-skeleton_gradle/order-service/src/main/java/com/example/order/service/OrderService.java
package com.example.order.service;

import java.time.Instant;
import java.util.List;

import org.springframework.stereotype.Service;

import com.example.order.domain.CreateOrderRequest;
import com.example.order.entity.OrderEntity;
import com.example.order.repo.OrderRepository;

@Service
public class OrderService {

    private final OrderRepository repo;

    public OrderService(OrderRepository repo) {
        this.repo = repo;
    }

    public OrderEntity create(CreateOrderRequest req) {
        return repo.save(new OrderEntity(req.userId(), req.symbol(), req.quantity(), Instant.now()));
    }

    public List<OrderEntity> listByUser(long userId) {
        return repo.findByUserId(userId);
    }
}

File: code-playbook/_kotlin/example_kotlin/src/main/kotlin/com/revy/service/OrderService.kt
package com.revy.service

import com.revy.domain.entity.Order
import jakarta.persistence.EntityManager
import org.springframework.stereotype.Service

@Service
class OrderService(
    val entityManager: EntityManager
) {

    fun createOrder(): Order {
        var newOrder = Order();
        entityManager.persist(newOrder)
        return newOrder;
    }

}
